scripts/FigS4A-B_celltype-diagnostic-plots.R

```diff
@@ -0,0 +1,199 @@
+
+# read in sce objects 
+processed_sce <- readr::read_rds(processed_sce_file)
+
+# define output file paths 
+plots_dir <- here::here("figures", "pngs") 
+singler_diagnostic_plot_file <- file.path(plots_dir, "FigS4A_singler-diagnostic.png")
+cellassign_diagnostic_plot_file <- file.path(plots_dir, "FigS4B_cellassign-diagnostic.png")
+
+# source in helper functions for plotting
+function_file <- here::here("scripts", "utils", "celltype-plot-helper-functions.R")
+source(function_file)
+
+# Create data frame of cell types
+celltype_df <- create_celltype_df(processed_sce)
+
+# Prep SingleR data for plotting -------------------------------------------------------
+
+# extract scores into matrix
+singler_scores <- metadata(processed_sce)$singler_result$scores
+
+# Create data frame for plotting with delta median and the full *non-pruned* cell labels
+delta_median_df <- tibble::tibble(
+  delta_median = rowMaxs(singler_scores) - rowMedians(singler_scores),
+  # Need to grab the non-pruned label for this plot
+  full_labels = metadata(processed_sce)$singler_result$labels,
+  # if pruned.labels are NA ==> low confidence
+  # so, negate for this variable:
+  confident = !is.na(metadata(processed_sce)$singler_result$pruned.labels)
+) |>
+  dplyr::mutate(
+    confident = ifelse(confident, "High-quality", "Low-quality")
+  )
+
+
+# we use inner_join b/c the above tibble does NOT contain "Unknown cell type", which
+#  we do not want to display here
+delta_median_df <- delta_median_df |>
+  dplyr::inner_join(
+    tibble::tibble(
+      full_labels = celltype_df$singler_celltype_ontology,
+      celltype = celltype_df$singler_celltype_annotation
+    ) |> dplyr::distinct()
+  ) |>
+  dplyr::select(-full_labels) |>
+  # filter out celltypes that have a low number of cells 
+  dplyr::add_count(celltype) |>
+  dplyr::filter(n >= 10)
+
+# add column with ordered levels with wrapped labels for visualization
+delta_median_df$annotation_wrapped <- factor(
+  delta_median_df$celltype,
+  # rev() so large groups are at the TOP of the plot
+  levels = rev(levels(delta_median_df$celltype)),
+  labels = rev(stringr::str_wrap(levels(delta_median_df$celltype), 30))
+)
+
+# Subset the data to just confident points for median+/-IQR
+delta_median_confident_df <- delta_median_df |> 
+  dplyr::filter(confident == "High-quality")
+
+# SingleR diagnostic plot ------------------------------------------------------
+
+# Plot delta_median across celltypes colored by pruning
+singler_diagnostic_plot <- ggplot(delta_median_df) +
+  aes(
+    x = delta_median,
+    y = annotation_wrapped,
+    shape = confident,
+    alpha = confident
+  ) +
+  ggforce::geom_sina(
+    size = 0.8,
+    color = "black", # will get applied to all confident points and non-confident outline
+    fill = "white", # will apply to non-confident fill only
+    position = position_dodge(width = 0.05) # Keep both types of points mostly in line
+  ) +
```

Comment:
If we go with the filtered version, maybe we can tweak something here. See how the low confidence points have a wider y-range? They don't look quite along the line anymore. I think the `maxwidth` or `bandwidth` args would be good to try out - https://ggforce.data-imaginist.com/reference/geom_sina.html